tasks/broad/RNAWithUMIsTasks.wdl

```diff
@@ -1,617 +1,628 @@
-            --read-structure ~{read2Structure} \
-            --molecular-index-tags RX \
-            --output extractUMIs.out.bam
-    >>>
-
-    runtime {
-        docker : docker
-        cpu : cpu
-        memory : "${memory_mb} MiB"
-        disks : "local-disk ${disk_size_gb} HDD"
-        preemptible: 0
-    }
+  command <<<
+    java -jar /usr/gitc/fgbio.jar ExtractUmisFromBam \
+      --input ~{bam} \
+      --read-structure ~{read1Structure} \
+      --read-structure ~{read2Structure} \
+      --molecular-index-tags RX \
+      --output extractUMIs.out.bam
+  >>>
 
-    output {
-        File bam_umis_extracted = "extractUMIs.out.bam"
-    }
+  runtime {
+    docker: docker
+    cpu: cpu
+    memory: "${memory_mb} MiB"
+    disks: "local-disk ${disk_size_gb} HDD"
+    preemptible: 0
+  }
+
+  output {
+    File bam_umis_extracted = "extractUMIs.out.bam"
+  }
 }
 
 task STAR {
-    input {
-        File bam
-        File starIndex
-
-        String docker = "us.gcr.io/broad-gotc-prod/samtools-star:1.0.0-1.11-2.7.10a-1642556627"
-        Int cpu = 8
-        Int memory_mb = 64000
-        Int disk_size_gb = ceil(2.2 * size(bam, "GB") + size(starIndex, "GB")) + 250
-    }
+  input {
+    File bam
+    File starIndex
 
-    command <<<
-        echo $(date +"[%b %d %H:%M:%S] Extracting STAR index")
-        mkdir star_index
-        tar -xvf ~{starIndex} -C star_index --strip-components=1
-
-        STAR \
-        --runMode alignReads \
-        --runThreadN 8 \
-        --genomeDir star_index \
-        --outSAMtype BAM Unsorted  \
-        --readFilesIn ~{bam} \
-        --readFilesType SAM PE \
-        --readFilesCommand samtools view -h \
-        --limitSjdbInsertNsj 1200000 \
-        --outSAMstrandField intronMotif \
-        --outSAMunmapped Within \
-        --outFilterType BySJout \
-        --outFilterMultimapNmax 20 \
-        --outFilterScoreMinOverLread 0.33 \
-        --outFilterMatchNminOverLread 0.33 \
-        --outFilterMismatchNmax 999 \
-        --outFilterMismatchNoverLmax 0.1 \
-        --alignIntronMin 20 \
-        --alignIntronMax 1000000 \
-        --alignMatesGapMax 1000000 \
-        --alignSJoverhangMin 8 \
-        --alignSJDBoverhangMin 1 \
-        --alignSoftClipAtReferenceEnds Yes \
-        --chimSegmentMin 15 \
-        --chimMainSegmentMultNmax 1 \
-        --chimOutType WithinBAM SoftClip \
-        --chimOutJunctionFormat 0 \
-        --twopassMode Basic \
-        --quantMode TranscriptomeSAM \
-        --quantTranscriptomeBan Singleend
-    >>>
-
-    runtime {
-        # Note: this is 'us.gcr.io/tag-team-160914/neovax-tag-rnaseq:v1', just pulled into a location visible to warp tests
-        docker : docker
-        cpu : cpu
-        memory : "${memory_mb} MiB"
-        disks : "local-disk ${disk_size_gb} HDD"
-        preemptible: 0
-    }
+    String docker = "us.gcr.io/broad-gotc-prod/samtools-star:1.0.0-1.11-2.7.10a-1642556627"
+    Int cpu = 8
+    Int memory_mb = 64000
```

Comment:
64 Gib seems like a lot here. Should this scale with input size or is a large static amount actually most appropriate?

Reply:
It is a lot. Let me look into making it scale.

Reply:
I've updated this to scale (but I don't think it makes a big difference)

Reply:
Sounds good!